crates/milli/src/vector/mod.rs

```diff
@@ -542,6 +543,46 @@ pub enum Embedder {
     Composite(composite::Embedder),
 }
 
+#[derive(Debug)]
+struct EmbeddingCache {
+    data: Option<Mutex<lru::LruCache<String, Embedding>>>,
+}
+
+impl EmbeddingCache {
+    const MAX_TEXT_LEN: usize = 2000;
+
+    pub fn new(cap: u16) -> Self {
+        let data = (|| Some(Mutex::new(lru::LruCache::new(NonZeroUsize::new(cap as usize)?))))();
+        Self { data }
+    }
```

Comment:
Why is it possible that it's None and why all this complexity around creating the Mutexed lru?

Reply:
Wow! Is it just to return None if the `NonZeroUsize` returns `None`?
```suggestion
    data: Option<Mutex<lru::LruCache<String, Embedding>>>,
}

impl EmbeddingCache {
    const MAX_TEXT_LEN: usize = 2000;

    pub fn new(cap: u16) -> Self {
        Self { data: NonZeroUsize::new(cap as usize).map(|cap| Mutex::new(lru::LruCache::new(cap))) }
    }
```

Reply:
Or even this would be more clear than all those parentheses 😂
```suggestion
    data: Option<Mutex<lru::LruCache<String, Embedding>>>,
}

impl EmbeddingCache {
    const MAX_TEXT_LEN: usize = 2000;

    pub fn new(cap: u16) -> Self 
        Self { data: NonZeroUsize::new(cap as usize).map(lru::LruCache::new).map(Mutex::new) }
    }
```